### PR TITLE
rename flash_attn_raw to flash_attn_unpadded

### PR DESCRIPTION
**Pull request by kuizhiqing**
### PR types

New features

### PR changes

APIs

### Describe

This pr include the following two modifs:
* rename `flash_attn_raw` to `flash_attn_unpadded` and their relatives
* add python api `flash_attn_unpadded` in `python/paddle/nn/functional/flash_attention.py`, but not in `__init__.py` yet.

### CODE REVIEW

File: python/paddle/nn/functional/flash_attention.py [hunk 1]
```diff
@@ -121,3 +121,131 @@ def flash_attention(
...
+            cu_seqlens_q,
+            cu_seqlens_k,
+            max_seqlen_q,
+            max_seqlen_k,
+            scale,
+            dropout,
+            causal,
+            return_softmax,
+        )
+        return result_attention, result_softmax
```
Comment (zyfncg) [inline python/paddle/nn/functional/flash_attention.py]: result_softmax_lse和seed_offset不是python api的返回结果，如果是只给反向使用的话需要在yaml中配置为intermediate

File: python/paddle/nn/functional/flash_attention.py [hunk 2]
```diff
@@ -121,3 +121,131 @@ def flash_attention(
...
+        'cu_seqlens_k': cu_seqlens_k,
+    }
+    outputs = {
+        'out': out,
+        'softmax': softmax,
+        'softmax_lse': softmax_lse,
+        'seed_offset': seed_offset,
+    }
+    helper.append_op(
+        type='flash_attn',
```
Comment (zyfncg) [inline python/paddle/nn/functional/flash_attention.py]: flash_attn -> flash_attn_unpadded

File: python/paddle/fluid/tests/unittests/test_flash_attention.py [hunk 3]
```diff
@@ -66,9 +69,9 @@ def setUp(self):
         self.causal = False
         self.return_softmax = False
 
-    def test_raw(self):
+    def test_unpadded(self):
```
Comment (zyfncg) [inline python/paddle/fluid/tests/unittests/test_flash_attention.py:72]: 有静态图的单测监控吗？

File: python/paddle/nn/functional/flash_attention.py (hunk 1 above)
Comment (kuizhiqing) [reply python/paddle/nn/functional/flash_attention.py]: 已添加

File: python/paddle/nn/functional/flash_attention.py (hunk 2 above)
Comment (kuizhiqing) [reply python/paddle/nn/functional/flash_attention.py]: done

File: python/paddle/fluid/tests/unittests/test_flash_attention.py (hunk 3 above)
Comment (kuizhiqing) [reply python/paddle/fluid/tests/unittests/test_flash_attention.py:72]: done

**Review by zyfncg — APPROVED** [review]

**Review by wanghaoshuang — APPROVED** [review]

**Review by sneaxiy — APPROVED** [review]

**Pull request merged**